README.md
<p align="center">
  <a href="https://github.com/dlsucomet">
    <img alt="DoNotDisturb" src="https://cdn3.emoji.gg/emojis/7907_DND.png" width="200"/>
  </a>
</p>

This is the repository for the research group Do Not Disturb's Instant Messaging Application for their study on OSIs.

## 📜 Requirements

Make sure these libraries, packages, and extensions are installed

1. Node.js
   Use node version 16.18.0 for consistency. the package expo-cli doesn't support node versions greater than 16.x (Gallium) yet.

2. yarn
   yarn is used all throughout the project to install packages, npm can be used but not fit for production when you plan to fork this project.

   Install yarn

   ```shell
   npm install --global yarn
   ```

3. expo-cli
   Expo CLI is a command-line tool that is the primary interface between a developer and other Expo tools. You are going to use it for different tasks in the development life cycle of your project such as serving the project in development, viewing logs, opening the app on an emulator or a physical device, and so on.

   Install expo-cli

   ```
   yarn global add expo-cli
   ```

4. prettier
   prettier is a VS Code extension to format code on save to have consistency on all files.

## 🚀 Quick start

Navigate into your new site’s directory and start it up.

1. Clone this repository.

2. Install all dependencies.
   on the root directory of your project run this on your terminal

   ```shell
   yarn
   ```

3. Start the project.

   ```shell
   yarn start
   ```

## 🧙 Members and Advisers

Advisers

1. Mr. Ryan Austin Fernandez
2. Dr. Briane Paul Samson

Members

1. Cyril Ethan A. de Guzman
2. Rayden E. Lizan
3. Phoebe Clare L. Ong
4. Leana Hyacinth V. Rebong


CHANGELOG.md
# DoNotDisturb Changelog

<!--
Changelog Format
- [your name] [#issue number](link to issue) {issue title}

NOTE: all lowercase and no ending punctuation marks
-->

## Next release v1.0

- [CY] - [#84](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/84) app(hotfix): clear or delete bubble
- [CY] - [#81](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/81) app(feature): track friend default status and modal backend hotfix
- [CY] - [#80](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/80) app(navigation): see members screen to chat screen navigation
- [RY] - [#79](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/79) app(hotfix): fixed delete members and suggested members
- [RY] - [#78](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/78) app(hotfix): edit members screen updates members
- [CY] - [#77](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/77) app(feature): change status individual
- [CY] - [#74](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/74) app(feature): add bubble backend
- [CY] - [#66](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/66) app(feature): message box bubble indicator
- [RY] - [#77](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/75) app(feature): change status gcs popup
- [RY] - [#72](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/72) app(feature): see edit members
- [RY] - [#67](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/67) app(feature): edit group status
- [RY] - [#64](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/64) app(hotfix): bubble display fix
- [RY] - [#62](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/62) app(feature): status-for-specific-aud
- [RY] - [#60](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/60) app(feature): bubble-edit-backend
- [RY] - [#59](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/59) app(feature) bubble backend
- [RY] - [#58](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/58) app(feature) default status navigation
- [RY] - [#57](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/57) app(feature) default status screen backend
- [RY] - [#12](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/12) app(screen): default status screen
- [CY] - [#40](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/40) app(feature): chat info backend
- [PB] - [#51](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/51) app(feature): custom duration popup
- [LE] - [#49](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/49) app(screen): edit bubble
- [PB] - [#47](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/47) app(feature): reset popup
- [PB] - [#45](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/45) app(feature): group name popup
- [PB] - [#43](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/43) app(feature): delete popup
- [PB] - [#41](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/41) app(screen): create audience screen
- [CY] - [#38](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/38) app(feature): implement group chat
- [LE] - [#18](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/33) app(screen): chat info screen
- [CY] - [#22](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/22) app(feature): source message boxes from firestore
- [PB] - [#32](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/32) app(screen): add bubble screen
- [PB] - [#31](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/31) app(feature): settings
- [PB] - [#23](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/23) app(feature): quick access view modal
- [PB] - [#21](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/21) app(screen): set status for specific audience
- [LE] - [#16](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/16) app(screen): new message screen
- [CY] - [#18](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/18) app(screen): chat box screen
- [PB] - [#24](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/24) app(feature): bubble see all modal screen
- [PB] - [#25](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/25) app(feature): bubble screen
- [PB] - [#19](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/19) app(screen): bubble screen
- [PB] - [#13](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/13) app(screen): settings screen
- [CY] - [#11](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/11) app(screen): messages screen
- [CY] - [#9](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/9) app(feature): home bottom tabs navigation
- [CY] - [#7](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/7) app(feature): add dependencies to enable expo to publish the app online
- [CY] - [#5](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/5) app(feature): base project
- [CY] - [#3](https://github.com/cyril-deguzman/donotdisturb-instant-messenger/issues/3) markdown(clean): clean and update markdowns
- [CY] - [#0](https://github.com/cyril-deguzman/donotdisturb-instant-messenger) initial commit


src/screens/Signup.js
import React, { useState } from "react";
import {
  StyleSheet,
  Text,
  View,
  TextInput,
  Image,
  SafeAreaView,
  TouchableOpacity,
  StatusBar,
  Alert,
} from "react-native";
import { setDoc, doc, addDoc, collection } from "firebase/firestore";
import { createUserWithEmailAndPassword, updateProfile } from "firebase/auth";
import { auth, database } from "../../config/firebase";

const backImage = require("../assets/backImage.png");

export default function Signup({ navigation }) {
  const [name, setName] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  const onHandleSignup = async () => {
    if (email !== "" && password !== "" && name !== "") {
      try {
        await createUserWithEmailAndPassword(auth, email, password);
        await updateProfile(auth.currentUser, { displayName: name });

        const statusID = await addDoc(collection(database, "online_statuses"), {
          expiry: null,
          message: "",
          osi: "Open to Chat",
          toggleTime: false,
        });

        await setDoc(doc(database, "users", auth.currentUser.uid), {
          image: "",
          userID: auth.currentUser.uid,
          statusID,
          name,
        });
      } catch (err) {
        Alert.alert("Signup error", err.message);
      }
    }
  };

  return (
    <View style={styles.container}>
      <Image source={backImage} style={styles.backImage} />
      <View style={styles.whiteSheet} />
      <SafeAreaView style={styles.form}>
        <Text style={styles.title}>Sign Up</Text>
        <TextInput
          style={styles.input}
          placeholder="Enter name"
          autoCapitalize="none"
          textContentType="username"
          autoFocus={true}
          value={name}
          onChangeText={(text) => setName(text)}
        />
        <TextInput
          style={styles.input}
          placeholder="Enter email"
          autoCapitalize="none"
          keyboardType="email-address"
          textContentType="emailAddress"
          autoFocus={true}
          value={email}
          onChangeText={(text) => setEmail(text)}
        />
        <TextInput
          style={styles.input}
          placeholder="Enter password"
          autoCapitalize="none"
          autoCorrect={false}
          secureTextEntry={true}
          textContentType="password"
          value={password}
          onChangeText={(text) => setPassword(text)}
        />
        <TouchableOpacity style={styles.button} onPress={onHandleSignup}>
          <Text style={{ fontWeight: "bold", color: "#fff", fontSize: 18 }}>
            {" "}
            Sign Up
          </Text>
        </TouchableOpacity>
        <View
          style={{
            marginTop: 20,
            flexDirection: "row",
            alignItems: "center",
            alignSelf: "center",
          }}
        >
          <Text style={{ color: "gray", fontWeight: "600", fontSize: 14 }}>
            Already have an account?{" "}
          </Text>
          <TouchableOpacity onPress={() => navigation.navigate("Login")}>
            <Text style={{ color: "#f57c00", fontWeight: "600", fontSize: 14 }}>
              {" "}
              Log in
            </Text>
          </TouchableOpacity>
        </View>
      </SafeAreaView>
      <StatusBar barStyle="light-content" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
  },
  title: {
    fontSize: 36,
    fontWeight: "bold",
    color: "orange",
    alignSelf: "center",
    paddingBottom: 24,
  },
  input: {
    backgroundColor: "#F6F7FB",
    height: 58,
    marginBottom: 20,
    fontSize: 16,
    borderRadius: 10,
    padding: 12,
  },
  backImage: {
    width: "100%",
    height: 340,
    position: "absolute",
    top: 0,
    resizeMode: "cover",
  },
  whiteSheet: {
    width: "100%",
    height: "75%",
    position: "absolute",
    bottom: 0,
    backgroundColor: "#fff",
    borderTopLeftRadius: 60,
  },
  form: {
    flex: 1,
    justifyContent: "center",
    marginHorizontal: 30,
  },
  button: {
    backgroundColor: "#f57c00",
    height: 58,
    borderRadius: 10,
    justifyContent: "center",
    alignItems: "center",
    marginTop: 40,
  },
});
